src/memory-cache.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2018 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ not
+ * use this file except in compliance with the License. You may obtain a copy
+ of
+ * the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ under
+ * the License.
+ */
+
+'use strict';
+
+import * as Koa from 'koa';
+
+type CacheEntry = {
+  saved: Date,
+  expires: Date,
+  headers: string,
+  payload: string,
+};
+
+export const CACHE_DURATION_MINUTES = 60 * 24;
+
+export class MemoryCache {
+  private store: {[id: string]: CacheEntry} = {};
+
+  async clearCache() {
+    this.store = {};
+  }
+
+  cacheContent(key: string, headers: {}, payload: Buffer) {
+    const now = new Date();
+
+    this.store[key] = {
+      saved: now,
+      expires: new Date(now.getTime() + CACHE_DURATION_MINUTES * 60 * 1000),
+      headers: JSON.stringify(headers),
+      payload: JSON.stringify(payload)
+    } as CacheEntry;
```

Comment:
You probably don't need the `as CacheEntry`, and it's safer to not cast because TypeScript lets you cast to a super or sub type without a warning, so you might e.g. accidentally miss a field.